nixos/modules/services/web-apps/immich.md

```diff
@@ -0,0 +1,277 @@
+be imported correctly.
+
+<!-- TODO: Is the following actually a Postgres bug? -->
+First, we need to fix the `search_path` variable of the dump, as a bug
+in PostgreSQL's `earthdistance` extension seems to cause issues when
+importing database dumps that contain its units.
+See [this StackOverflow comment](https://stackoverflow.com/questions/49702964/postgres-import-to-rds-could-not-execute-query-error-type-earth-does-not-ex#comment86441215_49702964)
+and [Immich "Backup and Restore" documentation](https://immich.app/docs/administration/backup-and-restore/)
+for more information.
+
+To work around this issue, we can adjust the `search_path` variable
+in the database dump. This variable is set within the first 20
+lines of your database dump, so you can just open up your database
+dump in your preferred text editor and apply the following changes:
+
+<!-- TODO: Is there a more user friendly way to display this change than using diff? -->
+```diff
+--- a/immich.sql
++++ b/immich.sql
+@@ -10,7 +10,7 @@ SET lock_timeout = 0;
+ SET idle_in_transaction_session_timeout = 0;
+ SET client_encoding = 'UTF8';
+ SET standard_conforming_strings = on;
+-SELECT pg_catalog.set_config('search_path', '', false);
++SELECT pg_catalog.set_config('search_path', 'public, pg_catalog', false);
+ SET check_function_bodies = false;
+ SET xmloption = content;
+ SET client_min_messages = warning;
+```
+
+The next issue we will need to work around is that Immich' Docker
+deployment caused all library files to be referenced relative to its
+working directory (`/usr/src/app` inside the container). This means
+that all photos, videos and thumbnails are going to appear with the
+path prefix `upload/` in the database dump, as Immich stores its data
+in `./upload` relative to its working directory by default.
+See [Immich "Files Custom Locations" documentation](https://immich.app/docs/guides/custom-locations/)
+for more information.
+In summary, the goal is to replace the `upload/` prefix with the new
+*absolute* media location path `TARGET_LOCATION`.
+
+In order to change these file paths in the database dump, we can use
+the following `sed` command:
+
+```console
+# sed "s#upload/(.*)/#$TARGET_LOCATION/\1/#g" ./immich.sql > ./immich-new-path.sql
+```
+
+This `sed` command makes use of capture groups as just replacing every
+occurrence of `upload/` would break paths referencing
+`upload/upload/`. In order to replace just the prefix, and not all
+occurrences of `upload` the left-hand-side also captures the path
+segment after `upload/`.
+
+### Prepare new database and library {#module-immich-migrating-from-docker-compose-prepare-db}
+
+To ensure that the new database does not contain any residual data
+from the brief time Immich was started on the system, we are going to
+drop all existing data. To do so, first start an interactive `psql`
+session as the Postgres superuser in the `immich` database:
+
+```console
+# sudo -u postgres psql immich
```

Comment:
Shouldn't we be running this as the immich user so set up permissions correctly from the get-go?

```suggestion
# sudo -u immich psql immich
```

Reply:
I *think* some of the extensions couldn't be created when I tried as `immich`.